app/services/one_off/create_or_update_statements.rb

```diff
@@ -0,0 +1,84 @@
+module OneOff
+  class CreateOrUpdateStatements
+    YEAR = {
+      "22" => 2022,
+      "23" => 2023,
+      "24" => 2024,
+      "25" => 2025,
+    }.freeze
+
+    MONTH = {
+      "Jan" => 1,
+      "Feb" => 2,
+      "Mar" => 3,
+      "Apr" => 4,
+      "May" => 5,
+      "Jun" => 6,
+      "Jul" => 7,
+      "Aug" => 8,
+      "Sep" => 9,
+      "Oct" => 10,
+      "Nov" => 11,
+      "Dec" => 12,
+    }.freeze
+    TEMPLATE_STATEMENT_YEAR = 2025
+    TEMPLATE_STATEMENT_MONTH = 1
+    def call(cohort_year:, csv_path:)
+      csv_file = CSV.read(csv_path, headers: true)
+
+      ActiveRecord::Base.transaction do
+        @cohort = Cohort.find_by!(start_year: cohort_year)
+
+        csv_file.each do |row|
+          LeadProvider.where.not(name: "National Institute of Teaching").find_each do |lead_provider|
+            year, month = parse_date(row["name"])
+            statements = Statement.where(year:, month:, cohort: @cohort, lead_provider: lead_provider)
+            return ArgumentError if statements.count > 1
+
+            statement = statements.first
+            if statement
+              statement.update!(output_fee: output_fee(row["output_fee"]))
+              FinancialChangeLog.log!(description: FinancialChangeLog::ONE_OFF_2326, data: { updated_statement_id: statement.id })
+            else
```

Comment:
```suggestion
              statement.update!(output_fee: output_fee(row["output_fee"]))
              FinancialChangeLog.log!(description: FinancialChangeLog::ONE_OFF_2326, data: { updated_statement_id: statement.id, , changes: statement.changes })
            else
```

things this change does:
- logs the data being changed
- only logs if the the output fee actually changes

wdyt?

Reply:
great suggestion

Reply:
Its not working as expected, we can investigate later - possibly `changes` gets reset when saved. With ticket number and the file available in the ticket we can easily track the change.

Reply:
yeah - `changes` and `changed` are reset when saved.
you would have to update the attribute, then check for `changed?`, and then log and save.

Reply:
Thats an idea - as we are less likely to miss the important changes. How about we can do it in the next iteration?